File: python/fastapi/src/app/modules/branchs/module.py
from app.common.classes.base_module import BaseModule

# Controller
from .controller import BranchsController
# Service
from .service import BranchsService
# Repository
from .repository import BranchsRepository

# Middlewares
from ..auth.middlewares.auth import auth_middleware

class BranchsModule(BaseModule):
    def __init__(self):
        _repository = BranchsRepository()
        _service = BranchsService(_repository)
        self._controller = BranchsController(_service)
        super().__init__(prefix="agencias")

    def _routes(self):
        self._router.get(self._controller.find_all, dependencies=[auth_middleware])

File: go/fiber_orm/internal/modules/branch/module.go
package branch

import (
	auth "tcc/internal/modules/auth/middleware"

	"github.com/gofiber/fiber/v2"
)

func BranchModule(router *fiber.App) {
	controller := BranchController{
		service: BranchService{
			repository: BranchRepository{},
		},
	}

	group := router.Group("/agencias")
	{
		group.Get("", auth.AuthMiddleware(), controller.FindAll)
	}
}